### PR TITLE
Unit tests in random order to make sure that unit tests are independent of each other

### PR DESCRIPTION
**Pull request by GernotMaier**
We had a couple of times the issue that unit tests depend on the order of execution. This is obviously wrong, as unit tests by definition should be testing 'units'.

This PR adds a CI step for the nightly test of main and runs the unit tests 5x in random order.

### CODE REVIEW

**Comment by GernotMaier** [comment]
Interestingly, it actually fails before correcting the tests in test_general.py, see https://github.com/gammasim/simtools/actions/runs/9107587029/job/25036730200

I removed this test, as this one tests for an executable without giving the full path. This is not what the function to be tested is supposed to handle.

**Comment by GernotMaier** [comment]
One more failure fixed: `test_model_parameter` was modifying a parameter of a TelescopeModel given by a fixture. This was used by other tests, which failed depending on the test sequence. Fixed now (and this is exactly why this PR is important).

**Comment by GernotMaier** [comment]
There is one last test failing: `tests/unit_tests/model/test_array_model.py::test_get_single_telescope_info_from_array_config`

This has been fixed in PR #910 and I will wait for this PR to be merged. 

(leave current PR in draft in the mean while)

File: tests/unit_tests/model/test_model_parameter.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def test_get_parameter_dict(telescope_model_lst):
 
 
 def test_get_parameter_value(telescope_model_lst):
-    tel_model = telescope_model_lst
+    tel_model = copy.deepcopy(telescope_model_lst)
```
Comment (VictorBarbosaMartins) [inline tests/unit_tests/model/test_model_parameter.py:35]: We usually uses copy.copy but I checked now in the reference API and deepcopy is probably better. Should we worry that `copy.copy` is not doing what we think it is in other parts of the code?

File: .github/workflows/CI-unittests.yml [hunk 2]
```diff
@@ -101,6 +101,12 @@ jobs:
         run: |
           pytest --durations=10 --color=yes -n auto --dist loadscope
 
+      - name: Random order
```
Comment (VictorBarbosaMartins) [inline .github/workflows/CI-unittests.yml:104]: One question: is this going to be applied only to the main or to each branch?

**Review by VictorBarbosaMartins — APPROVED** [review]
Thanks. Good to have this ensuring the tests are not depending on each other.

File: tests/unit_tests/model/test_model_parameter.py (hunk 1 above)
Comment (GernotMaier) [reply tests/unit_tests/model/test_model_parameter.py:35]: My assumption is always that if the test does not fail, then `copy.copy` is enough. In this case, we required a deepcopy.

File: .github/workflows/CI-unittests.yml (hunk 2 above)
Comment (GernotMaier) [reply .github/workflows/CI-unittests.yml:104]: Yes, only to main. In 99% of all cases, tests don't depend on their sequence. For the rare cases for which they do, it is better good enough to find this out after merge-to-main.

To add: this is only used for the scheduled runs which runs automatically every night on the main branch.

**Comment by GernotMaier** [comment]
@VictorBarbosaMartins - thanks for the review!

**Pull request merged**